thousandeyes/data_source_thousandeyes_alert_rules.go

```diff
@@ -0,0 +1,179 @@
+				ConflictsWith: []string{"rule_name"},
+			},
+			"rule_name": {
+				Type:          schema.TypeString,
+				Optional:      true,
+				Description:   "Name of the alert rule",
+				ConflictsWith: []string{"rule_id"},
+			},
+			"expression": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "string expression of alert rule",
+			},
+			"direction": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "optional field with one of the following values: TO_TARGET, FROM_TARGET, BIDIRECTIONAL, for applicable alert types (eg. path trace, End-to-End (Agent) etc.)",
+			},
+			"notify_on_clear": {
+				Type:        schema.TypeInt,
+				Computed:    true,
+				Description: "1 to send notification when alert clears",
+			},
+			"default": {
+				Type:        schema.TypeInt,
+				Computed:    true,
+				Description: "Alert rules allow up to 1 alert rule to be selected as a default for each type. By checking the default option, this alert rule will be automatically included on subsequently created tests that test a metric used in alerting here",
+			},
+			"alert_type": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "type of alert rule, as determined by metric selection",
+			},
+			"minimum_sources": {
+				Type:        schema.TypeInt,
+				Computed:    true,
+				Description: "the minimum number of agents or monitors that must meet the specified criteria in order to trigger the alert",
+			},
+			"minimum_sources_pct": {
+				Type:        schema.TypeInt,
+				Computed:    true,
+				Description: "the minimum percentage of all assigned agents or monitors that must meet the specified criteria in order to trigger the alert",
+			},
+			"rounds_violating_out_of": {
+				Type:        schema.TypeInt,
+				Computed:    true,
+				Description: "applies to only v6 and higher, specifies the divisor (y value) for the “X of Y times” condition.",
+			},
+			"rounds_violating_required": {
+				Type:        schema.TypeInt,
+				Computed:    true,
+				Description: "applies to only v6 and higher, specifies the numerator (x value) for the X of Y times” condition",
+			},
+			"rounds_violating_mode": {
+				Type:     schema.TypeString,
+				Computed: true,
+				// `ANY` or `EXACT` / ANY
+				Description: "allows user to enforce a condition on alerts that will required the same agent to meet the condition(s) multiple rounds in a row, rather than any agents meeting the condition(s) ",
+			},
+		},
+	}
+}
+
+func dataSourceThousandeyesAlertRuleRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*thousandeyes.Client)
+
+	var found *thousandeyes.AlertRule
+
+	searchName := d.Get("rule_name").(string)
+	searchRuleID := d.Get("rule_id").(int)
+
+	log.Printf("[INFO] ###### Name: %s | ID: %d", searchName, searchRuleID)
+
+	alertRules, err := client.GetAlertRules()
+	if err != nil {
+		return err
+	}
+
+	if searchName != "" {
```

Comment:
```go
	for _, agent := range *agents {
		if ar.RuleName == searchRuleName {
			found = &agent
			break
		}
                if ar.RuleID == searchRuleID {
                        found = &agent
                        break
	}
```

Can we simplify this to something like this? 

Or should we even support adding the ID into the search? If you have the ID you can use it manually in your test?

Reply:
I think we probably should support both to be honest, especially in my use case where there will likely be a transition period to TF from Manual config... I do like this suggestion... seems more idiomatic them my hacky approach.